libraries/core_libs/network/src/tarcap/packets_handlers/votes_sync_packet_handler.cpp

```diff
@@ -35,21 +35,37 @@ void VotesSyncPacketHandler::process(const PacketData &packet_data, const std::s
     return;
   }
 
+  std::map<blk_hash_t, std::pair<uint64_t, uint64_t>> voted_value_round_step;
+
   std::vector<std::shared_ptr<Vote>> next_votes;
   const auto next_votes_count = packet_data.rlp_.itemCount();
   for (size_t i = 0; i < next_votes_count; i++) {
     auto next_vote = std::make_shared<Vote>(packet_data.rlp_[i].data().toBytes());
     const auto next_vote_hash = next_vote->getHash();
+
+    // votes for the same value should be from one round and step
+    auto [previous_votes_round, previous_votes_step] = voted_value_round_step[next_vote->getBlockHash()];
```

Comment:
I do not understand this logic. First you check here but you insert few lines under? so is this check even working?

Reply:
We could have votes for 2 values here. One of this values should be NULL_HASH. So yeah, I missed one more check here